### PR TITLE
docs: Support GFM style admonitions in option descriptions

### LINKED ISSUE #2217: docs: use consistent admonitions syntax in options and user-guides
- [x] #2213
- [x] #2218
- [ ] #2221

#2213 added support for rendering github-flavoured "markdown alerts":

```md
> [!TIP]
> This is a tip
```

Using the mdbook-alerts preprocessor, we now renders this as:

```html
<div class="mdbook-alerts mdbook-alerts-tip">
  <p class="mdbook-alerts-title">
    <span class="mdbook-alerts-icon"></span>
    tip
  </p>
  <p>This is a tip</p>
</div>
```

> [!TIP]
> This is a tip

---

However, we render our option descriptions using [`nixos-render-docs`](https://github.com/NixOS/nixpkgs/blob/1376d89f1394fd5111a7306a3e72c8f9359e36d8/pkgs/tools/nix/nixos-render-docs/default.nix#L21), via the [`optionsCommonMark` function](https://github.com/NixOS/nixpkgs/blob/1376d89f1394fd5111a7306a3e72c8f9359e36d8/nixos/lib/make-options-doc/default.nix#L182-L190).
https://github.com/nix-community/nixvim/blob/e864f2c9e0cda66b0f419b2c3e36374c177d3712/docs%2Fmdbook%2Fdefault.nix#L13-L18

This takes an `options.json` file parses its markdown to an AST and spits out commonmark, asciidoc, or manpages (as requested).

Because it uses an abstract syntax tree, recognized-but-invalid symbols (such as `[foo]` where there's no corresponding `[foo]: URL`) end up being escaped (`\[foo\]`) in the resulting commonmark output.

It's also probably not a good idea to use syntax not supported by nixos-render-docs in option descriptions, since it will render poorly in the manpage docs.

---

As documented [here](https://github.com/NixOS/nixpkgs/blob/master/doc/README.md#admonitions), nixpkgs have adopted an alternative syntax for admonitions:

```md
::: {.warning}
This is a warning
:::
```

This is based on pandoc’s [fenced `div`s syntax](https://github.com/jgm/commonmark-hs/blob/master/commonmark-extensions/test/fenced_divs.md) which is _intended_ to be used to add HTML classes or IDs to the rendered `div`.

`nixos-render-docs options commonmark` will render the above as `**Warning:** This is a warning`, which mdbook then renders as:

```html
<p><strong>Warning:</strong> This is a warning</p>
```
**Warning:** This is a warning

[Implemented here](https://github.com/NixOS/nixpkgs/blob/8ae535d44c1b8002cf53c3456811306b6834cd6f/pkgs/tools/nix/nixos-render-docs/src/nixos_render_docs/commonmark.py#L46-L52).

### PR DESCRIPTION
**Pull request by traxys**
Follow up to https://github.com/nix-community/nixvim/pull/2213 & https://github.com/nix-community/nixvim/pull/2218. Fixes #2217.

This is based on https://github.com/antfu/markdown-it-github-alerts/blob/23d2d84c738e206b74762e388ca35c076d170028/src/index.ts#L61-L88.

### CODE REVIEW

File: docs/default.nix
Comment (MattSturgeon) [inline docs/default.nix:44]: Nit: ideally the patch should be generated on top of the other patch, and then this would be `0002`

File: docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch
Comment (MattSturgeon) [inline docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch:26]: This function is large enough that it makes me uncomfortable having it directly in a patch file.

I think it'll end up being a maintenance nightmare if we ever want to work on it.

I'm tempted to suggest extracting it into either a loose file or a nix string variable.

If we did that, we could add a `@PLACEHOLDER@` to the patch, then substitute in the actual function in `postPatch`.

**Review by MattSturgeon — COMMENTED** [review]
I'm not familiar enough with `nixos-render-docs` to spot any footguns, but if the patched tests are passing then we should be good. Thanks for tackling this so quickly, I wasn't looking forward to it!

Have you tested out these changes?

What do you think of the suggestions below? I think `_gfm_alert` defined in a loose file that is injected using `substituteInPlace` may be more maintainable; I always struggle working directly with patch files 😅 

It'd be nice to have some comments in the implementation too, and I think a loose file would make adding/editing/reading them easier.

What do you think?

File: docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch
Comment (MattSturgeon) [reply docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch:26]: Or maybe I just need to re-read the patch on a larger screen and with fresh eyes tomorrow.

Just using a patch is certainly simpler than having a patch, a loose file, _and_ a substitution...

**Comment by MattSturgeon** [comment]
> This is based on [antfu/markdown-it-github-alerts@`23d2d84`/src/index.ts#L61-L88](https://github.com/antfu/markdown-it-github-alerts/blob/23d2d84c738e206b74762e388ca35c076d170028/src/index.ts#L61-L88)

I'm also looking at https://github.com/ByPikod/markdown-it-github-alert/blob/f17613451396d7aa43cf6e8a3a81a4663ff1f27b/src/plugin.ts#L72-L122 to see compare the two

File: docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch
Comment (traxys) [reply docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch:26]: (in term of simplicity)

**Comment by traxys** [comment]
I tested on a plugin that it worked, though I only tested with a  `TIP`

**Comment by MattSturgeon** [comment]
> > This is based on [antfu/markdown-it-github-alerts@`23d2d84`/src/index.ts#L61-L88](https://github.com/antfu/markdown-it-github-alerts/blob/23d2d84c738e206b74762e388ca35c076d170028/src/index.ts#L61-L88)
> 
> I'm also looking at [ByPikod/markdown-it-github-alert@`f176134`/src/plugin.ts#L72-L122](https://github.com/ByPikod/markdown-it-github-alert/blob/f17613451396d7aa43cf6e8a3a81a4663ff1f27b/src/plugin.ts#L72-L122) to see compare the two

The other implementation leverages the "actual" blockquote rule, extracted from MarkdownIt. I wonder if that makes it more robust? [L65-L69](https://github.com/ByPikod/markdown-it-github-alert/blob/f17613451396d7aa43cf6e8a3a81a4663ff1f27b/src/plugin.ts#L65-L69)

I also like having `matchAlertType` in a separate function (returns `null` or "type"), see: [parser.ts](https://github.com/ByPikod/markdown-it-github-alert/blob/f17613451396d7aa43cf6e8a3a81a4663ff1f27b/src/parser.ts#L13). I think this makes things more readable in the end and allows for slightly more robust parsing (see some of the comments in that file).

Using named regex groups may also help to that end. Python uses `(?P<name>.*)` syntax. You then use either `m.groupdict()` or `m.group('name')` instead of `m.group(0)`.

**Comment by traxys** [comment]
I agree that the implementation you linked seemed a bit more robust, but we should be _ok_ as nested GFM alerts are not permitted.

I skimmed through the nixos-render-docs implementation and they _seem_ to use the same level of rules for their handling of `::: {.warning}`.

File: docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch
Comment (MattSturgeon) [inline docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch:45]: I like the idea of splitting this into another function, but it isn't that important.

File: docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch
Comment (MattSturgeon) [inline docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch:47]:
```suggestion
+                # Remove "> [!TIP]\n" from the token
+                first_content.content = first_content.content[m.endpos:].lstrip()
```

- Let's use `Match.endpos` to get the substring index
- Let's add a comment so this makes sense in 12 months time :upside_down_face: 
- Do we really need `lstrip()`? I think it would be better to match whitespace in the regex.

File: docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch
Comment (MattSturgeon) [inline docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch:35]: Lets add a comment, maybe `Look for the start/end of the blockquote`? Maybe also clarify it relates to _tokens_ not chars? I.e. we are dealing with a list of `Token` objects.

File: docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch
Comment (MattSturgeon) [inline docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch:42]: I don't fully understand this, so I think it needs some comments.

It looks for the first "inline" type token in the pre-tokenized blockquote? Using the index-bounds we searched for above.

File: docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch
Comment (MattSturgeon) [inline docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch:50]: Nit: let's be explicit about this being a regex group:
```suggestion
+                open.meta['kind'] = m.group(1).lower()
```

File: docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch
Comment (MattSturgeon) [inline docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch:52]: I don't see us declaring or reading `open` and `close` anywhere, so I assume they are coming from MarkdownIt. Are we inheriting these from a super class?

(forgive my lack of python-foo)

File: docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch
Comment (MattSturgeon) [inline docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch:55]: I wouldn't be against inlining `gfm_alert_to_adm` to an anymous lambda.

File: docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch
Comment (MattSturgeon) [reply docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch:52]: Let's also add a comment here, stating what we're doing here at the end of the function.

I guess something like `set the open & close class properties in the same way as nixpkgs admonitions`

**Review by MattSturgeon — APPROVED** [review]
I have some suggestions for making this more readable (below) but overall I'm happy, so here's my green tick :grinning: 

Awesome work again @traxys :tada:

File: docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch
Comment (traxys) [reply docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch:42]: Yeah exactly, it mimics a JS `slice(....).find(....)`

Comment (MattSturgeon) [reply docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch:42]: I guess it makes me wonder what non-inline token types it may run into. And whether any of them should trigger errors or an early return.

Comment (traxys) [reply docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch:42]: I copied this behaviour from the project I linked, I don't know a lot about markdown-it nor python

File: docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch
Comment (traxys) [reply docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch]: What part would you want in another function?

File: docs/0002-nixos-render-docs-Support-gfm-style-admonitions.patch
Comment (MattSturgeon) [inline docs/0002-nixos-render-docs-Support-gfm-style-admonitions.patch:4]: I know this is unimportant, but:

Since this is now 2/2 can you update my patch too to be part of the patchset?

If you do that, maybe reword one of the commits so either none or both use the `nixos-render-docs: ` prefix?

File: docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch
Comment (traxys) [reply docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch]: `open = <.....>` declares the variable at the start of the loop

File: docs/gfm-alerts-to-admonitions/default.nix
Comment (MattSturgeon) [inline docs/gfm-alerts-to-admonitions/default.nix:14]: This probably doesn't really matter in nixvim, but this is frowned upon because changes to _anything_ in `./.` will trigger a rebuild, even stuff in _this file_ that'd normally just be metadata or passthru.

File: docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch
Comment (traxys) [reply docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch]: I don't like those very much :/ I find the current approach readable enough.
I also took other plugins as examples, they were all structured like this (in nixos-render-docs or in markdown it plugins)

File: docs/default.nix
Comment (MattSturgeon) [inline docs/default.nix:40]: I know this is kinda off-topic, but I can never wrap my head around when to use native or not and propagated or not for buildInputs...

My understanding is:
- buildInputs generally only relate to the drv builder, not the package's runtime
- `native` buildInputs are for exes that should be run from the PATH during the build
- `propagated` buildInputs are for deps that should also be available to builders of _other packages_ that include this package in _their_ buildInputs (i.e. transitives get propagated)
- propagatedNativeBuildInputs is also a thing, see above two

Here we shouldn't need propagated right?

File: docs/default.nix
Comment (MattSturgeon) [inline docs/default.nix:44]: I don't fully understand why we need to override `doCheck` here? Especially since we aren't overriding/overlaying `markdown-it-py` anywhere :face_with_diagonal_mouth:

File: docs/gfm-alerts-to-admonitions/gfm_alerts_to_admonitions/main.py
Comment (MattSturgeon) [inline docs/gfm-alerts-to-admonitions/gfm_alerts_to_admonitions/main.py:1]: Much more maintainable :heart_eyes:

File: docs/gfm-alerts-to-admonitions/tests/test_plugin.py
Comment (MattSturgeon) [inline docs/gfm-alerts-to-admonitions/tests/test_plugin.py:1]: We even have tests :astonished:

File: docs/gfm-alerts-to-admonitions/tests/test_plugin.py
Comment (MattSturgeon) [inline docs/gfm-alerts-to-admonitions/tests/test_plugin.py:8]: `upper` shouldn't be needed, GFM-alerts are case-insensitive.

Maybe we should test mixed cases too?

File: flake-modules/dev/default.nix
Comment (MattSturgeon) [inline flake-modules/dev/default.nix:55]: Any reason why? (we have a yaml formatter configured IIRC)

**Review by MattSturgeon — COMMENTED** [review]
Awesome :tada:

File: docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch
Comment (traxys) [reply docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch]: `endpos` is not the same as `len(m[0])`:

> The value of endpos which was passed to the [search()](https://docs.python.org/3/library/re.html#re.Pattern.search) or > [match()](https://docs.python.org/3/library/re.html#re.Pattern.match) method of a [regex object](https://docs.python.org/3/library/re.html#re-objects). This is the index into the string beyond which the RE engine will not go.

In pratice here it is the length of the string

File: docs/0002-nixos-render-docs-Support-gfm-style-admonitions.patch
Comment (traxys) [reply docs/0002-nixos-render-docs-Support-gfm-style-admonitions.patch:4]: I don't know how to apply your patch in git, I use `patch -p1`, so it would reset all the metadata :/

File: docs/gfm-alerts-to-admonitions/default.nix
Comment (traxys) [reply docs/gfm-alerts-to-admonitions/default.nix:14]: Yeah, I saw how nixos-render-docs did it, but I did not want to bother with the complexity of filtering. I'm betting on us not touching this part very often

File: docs/default.nix
Comment (MattSturgeon) [inline docs/default.nix:110]: I wonder if `gfm-admonitions-to-nixpkgs` is a better name?

Admonitions is a generic term, but the tokens we're targeting are nixpkgs-sepecific.

File: docs/default.nix
Comment (traxys) [reply docs/default.nix:44]: We don't, but nixos-render-docs [does](https://github.com/NixOS/nixpkgs/blob/b94f2597143d383aeed61a8f036b596d34a26dfc/pkgs/tools/nix/nixos-render-docs/default.nix#L7-L17).

I fiddled with a number of ways to pass our dependency, this one was the only one which did not raise a double-dependency error

File: docs/gfm-alerts-to-admonitions/tests/test_plugin.py
Comment (traxys) [reply docs/gfm-alerts-to-admonitions/tests/test_plugin.py:8]: The regex _should_ match different cases, we could indeed add a test

File: flake-modules/dev/default.nix
Comment (traxys) [reply flake-modules/dev/default.nix:55]: `pytest-regression` performs a `diff`, so the formatter messes with it

Comment (MattSturgeon) [reply flake-modules/dev/default.nix:55]: Makes sense. Could you note that in a comment?

File: docs/default.nix
Comment (traxys) [reply docs/default.nix:110]: I feel like `nixpkgs` is too vague, it would have to be `nixos-render-docs` to be descriptive enough, but it's a bit long.

The plugin itself does not really rely on nixos-render-docs though, it creates `admonition_open` and `admoninition_close` tokens.
How to parse them is orthogonal to this plugin, so the name still makes sense for me

File: docs/gfm-alerts-to-admonitions/tests/test_plugin.py
Comment (MattSturgeon) [reply docs/gfm-alerts-to-admonitions/tests/test_plugin.py:8]: The implementation looks good, we probably don't need a test but it can't hurt.

File: docs/gfm-alerts-to-admonitions/gfm_alerts_to_admonitions/main.py
Comment (MattSturgeon) [inline docs/gfm-alerts-to-admonitions/gfm_alerts_to_admonitions/main.py:43]: Looks like you tried `m.endpos` but reverted. Did it not do what I thought it would?

Comment (traxys) [reply docs/gfm-alerts-to-admonitions/gfm_alerts_to_admonitions/main.py:43]: Yeah, I responded in the original suggestion with what it really does

Comment (traxys) [reply docs/gfm-alerts-to-admonitions/gfm_alerts_to_admonitions/main.py:43]: (I was glad to have the tests, as they caught the issue immediately)

File: docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch
Comment (MattSturgeon) [reply docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch]: Ah, I guess we want [`m.end(0)`](https://docs.python.org/3/library/re.html#re.Match.end) then. Sorry, just doing my best to read the docs and usually reading the wrong part :sweat_smile:

File: docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch
Comment (MattSturgeon) [reply docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch]: I was originally thinking we could be inspired by [parser.ts](https://github.com/ByPikod/markdown-it-github-alert/blob/f17613451396d7aa43cf6e8a3a81a4663ff1f27b/src/parser.ts#L13), which exports a "matcher" function returning either `null` or the matched type.

However, I think it is fine to keep this inline as it makes removing the matched substring easier.

File: docs/0002-nixos-render-docs-Support-gfm-style-admonitions.patch
Comment (MattSturgeon) [reply docs/0002-nixos-render-docs-Support-gfm-style-admonitions.patch:4]:
```
git am --directory pkgs/**/nixos-render-docs/src ~/nixvim/main/docs/0001-Output-GFM-admonition.patch
```

File: docs/default.nix
Comment (MattSturgeon) [reply docs/default.nix:44]: Is this because both our package and nixos-render-docs have a propagated dep on markdown-it-py? (sorry I'm still a bit confused)

Comment (traxys) [reply docs/default.nix:44]: Yeah exactly, and they end up being _not_ the same package due to their override

Comment (MattSturgeon) [reply docs/default.nix:44]: Maybe an alternative wording for the comments could be:

> Use the same override as `nixos-render-docs` does, to avoid "duplicate dependency" errors

File: docs/gfm-alerts-to-admonitions/default.nix
Comment (MattSturgeon) [reply docs/gfm-alerts-to-admonitions/default.nix:14]: Normally `./src` would be best practice, but as I said it shouldn't be too important for an internal package only used when we build our docs.

File: docs/gfm-alerts-to-admonitions/gfm_alerts_to_admonitions/__init__.py
Comment (MattSturgeon) [inline docs/gfm-alerts-to-admonitions/gfm_alerts_to_admonitions/__init__.py:1]: We're getting a linting error: https://buildbot.nix-community.org/#/builders/448/builds/1104/steps/1/logs/stdio

Comment (traxys) [reply docs/gfm-alerts-to-admonitions/gfm_alerts_to_admonitions/__init__.py:1]: I have no idea what that means xD

Comment (traxys) [reply docs/gfm-alerts-to-admonitions/gfm_alerts_to_admonitions/__init__.py:1]: I'll look it up

Comment (traxys) [reply docs/gfm-alerts-to-admonitions/gfm_alerts_to_admonitions/__init__.py]: See https://docs.astral.sh/ruff/rules/unused-import/ for more details

File: flake-modules/dev/default.nix
Comment (MattSturgeon) [inline flake-modules/dev/default.nix:57]: Seems to not be working. Maybe the leading `/` is incorrect?

https://buildbot.nix-community.org/#/builders/720/builds/1095/steps/1/logs/stdio

File: docs/gfm-alerts-to-admonitions/gfm_alerts_to_admonitions/main.py
Comment (MattSturgeon) [inline docs/gfm-alerts-to-admonitions/gfm_alerts_to_admonitions/main.py:11]: I think we can make this a little less verbose with some imports;

```py
from markdown_it import MarkdownIt
from markdown_it.rules_core import StateCore
```

Maybe we could also add type annotations for the `[Token]` list below? [I think](https://markdown-it-py.readthedocs.io/en/latest/api/markdown_it.token.html) that'd be `from markdown_it.token import Token`

File: docs/gfm-alerts-to-admonitions/default.nix
Comment (MattSturgeon) [reply docs/gfm-alerts-to-admonitions/default.nix:14]: To clarify, I wasn't talking about the `filter` function used with `lib.cleanSourceWith`.

I was referring to https://github.com/NixOS/nixpkgs/issues/301014:

> Derivations that use `src = ./.` will be rebuilt on any change to the `.nix` file that includes them, including formatting changes caused by `nixfmt-rfc-style` for [RFC 166](https://github.com/NixOS/rfcs/blob/master/rfcs/0166-nix-formatting.md?rgh-link-date=2024-04-02T18%3A12%3A28Z).

File: docs/default.nix
Comment (khaneliman) [reply docs/default.nix:40]: > I know this is kinda off-topic, but I can never wrap my head around when to use native or not and propagated or not for buildInputs...
> 
> My understanding is:
> 
>     * buildInputs generally only relate to the drv builder, not the package's runtime
> 
>     * `native` buildInputs are for exes that should be run from the PATH during the build
> 
>     * `propagated` buildInputs are for deps that should also be available to builders of _other packages_ that include this package in _their_ buildInputs (i.e. transitives get propagated)
> 
>     * propagatedNativeBuildInputs is also a thing, see above two
> 
> 
> Here we shouldn't need propagated right?

buildInputs are runtime available packages, actually.
native are build only
propogated do the same but to derivations that inherit it.

File: docs/0001-nixos-render-docs-Output-GFM-admonition.patch
Comment (khaneliman) [inline docs/0001-nixos-render-docs-Output-GFM-admonition.patch:47]: Not sure I understand what this version change does

File: docs/0002-nixos-render-docs-Support-gfm-style-admonitions.patch
Comment (khaneliman) [inline docs/0002-nixos-render-docs-Support-gfm-style-admonitions.patch:111]: is the version change just to match this? or could we just change it at this location?

File: docs/0001-nixos-render-docs-Output-GFM-admonition.patch
Comment (traxys) [reply docs/0001-nixos-render-docs-Output-GFM-admonition.patch:47]: @MattSturgeon and I woul have had different git versions, as I re-generated the patch so it would be 1/2

File: docs/0002-nixos-render-docs-Support-gfm-style-admonitions.patch
Comment (MattSturgeon) [reply docs/0002-nixos-render-docs-Support-gfm-style-admonitions.patch:111]: As said in https://github.com/nix-community/nixvim/pull/2221#discussion_r1758341074 it is just the git version used to run `git format-patch`.

I asked traxys to amend my prior patch so that a) it'd be `[1/2]` and b) it'd be worded consistently in the commit message.

I can push up some fixup commits with patch files generated by my git version to avoid the noise in the diff, if you like.

**Review by MattSturgeon — APPROVED** [review]
Apologies; I had to force-push as b61fae44991ea6c322985b6fdf45fa052485b15d would've been a no-op if committed at the HEAD of the PR.
EDIT: If you don't plan to merge immediately, I'll also take the liberty of rebasing after I merge #2229, since that'll conflict.

Once rebased & autosquashed this all looks good to me.

TBH, I wasn't convinced we'd ever get around to doing this and was making piece with using github-syntax in user-guides and nixpkgs-syntax in options... Thank you so much for not only tackling it, but also re-working it as a separately maintainable package derivation!

I'd love to have @GaetanLepage's eyes on this too, as he is far more experienced in python than me. But I don't want to put any pressure on him due to his other commitments, so it's up to you how long you will wait for additional feedback before merging.

**Review by khaneliman — APPROVED** [review]
LGTM, would be nice to have some docs about writing documentation in our contributing... but I understand that's always hard to have the energy/motivation to do, especially from scratch.

File: docs/0001-nixos-render-docs-Output-GFM-admonition.patch
Comment (khaneliman) [reply docs/0001-nixos-render-docs-Output-GFM-admonition.patch]: Ah, alright, I rarely look that closely at git patches. That makes sense.

**Comment by MattSturgeon** [comment]
> LGTM, would be nice to have some docs about writing documentation in our contributing...

I don't think we'd need much more than this on the topic:

```md
> [!TIP]
> You can use GitHub-style alerts within option descriptions if you need to emphasise something. e.g.
>
> ```md
> > [!WARNING]
> > This is a warning!
> ```
```

> [!TIP]
> You can use GitHub-style alerts within option descriptions if you need to emphasise something. e.g.
>
> ```md
> > [!WARNING]
> > This is a warning!
> ```

...But we'd presumably want it to be part of a wider section on style and syntax? So probably best in a follow up PR.

**Comment by khaneliman** [comment]
> > Would be nice to have some docs about writing documentation in our contributing...
> 
> I don't think we'd need much more than this on the topic:
> 
> ```
> > [!TIP]
> > You can use GitHub-style alerts within option descriptions if you need to emphasise something. e.g.
> >
> > ```md
> > > [!WARNING]
> > > This is a warning!
> > ```
> ```
> 
> Tip
> 
> You can use GitHub-style alerts within option descriptions if you need to emphasise something. e.g.
> 
> ```
> > [!WARNING]
> > This is a warning!
> ```
> 
> ...But we'd presumably want it to be part of a wider section on style and syntax? So probably best in a follow up PR.

Yeah, I wasn't very clear... I meant it'd be nice to have a quick bit about this functionality... but it might feel weird by itself without corresponding documentation on the documentation strategy, in general.

**Comment by traxys** [comment]
@mergifyio queue

**Pull request merged**
